### DISCUSSION TITLE
Livewire 2.5 - emitTo() doesn't work

### DISCUSSION THREAD

**Discussion by jadersbr** [livewire/livewire#4081] [1. Help]
Hi everyone,

I'm trying to use event emitTo(), but did not work. Seems the event is not listened by target component.

![image](https://user-images.githubusercontent.com/58928002/138528834-f8916f07-fe66-4d61-b634-42a83508339e.png)

**Button**

```
<button wire:click="sendEvent">
   Send event
</button>
```

**Main component**

```
public function sendEvent()
 {
        $this->emitTo('toolbar', 'eventTest', '123');
 }
```

**Target component**

```
class Toolbar extends Component
{
   protected $listeners = [
        'eventTest'                => 'test',
   ];

   public function test()
   {
        dd('toolbar test');
   }
```

What am I doing wrong? public function test is not fired.

If I use emit(), this works.

Laravel 7
Livewire 2.5

**Comment by devcircus** [comment]
Is the toolbar component nested in another folder?

Thanks,

Clay
________________________________
From: jadersilvabr ***@***.***>
Sent: Friday, October 22, 2021 5:20:36 PM
To: livewire/livewire ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [livewire/livewire] Livewire 2.5 - emitTo() doesn't work (Discussion #4081)


Hi everyone,

I'm trying to use event emitTo(), but did not work. Seems the event is not listened by target component.

[image]<https://user-images.githubusercontent.com/58928002/138528834-f8916f07-fe66-4d61-b634-42a83508339e.png>

Button

<button wire:click="sendEvent">
   Send event
</button>


Main component

public function sendEvent()
 {
        $this->emitTo('toolbar', 'eventTest', '123');
 }


Target component

class Toolbar extends Component
{
   protected $listeners = [
        'eventTest'                => 'test',
   ];

   public function test()
   {
        dd('toolbar test');
   }


What am I doing wrong? public function test is not fired.

If I use emit(), this works.

Laravel 7
Livewire 2.5

**Reply by jadersbr** [reply]
Yes, here a screenshot of my folder's structure:

![image](https://user-images.githubusercontent.com/58928002/138576238-4a98f6ce-d599-4a36-9ea3-111ab2c8121a.png)

DataTable2 is sending an event to Tollbar

**Reply by devcircus** [reply]
Just be sure to specify the path to the component in your "emitTo". Emit works because it's available to all components. 

```php
$this->emitTo('general.data-table.toolbar', 'my-event', $data);
```

Depending on how your view folders are named, you may need to tweak the path to the component, but that's the general idea.
